trestle/transforms/transformer_factory.py

```diff
@@ -46,6 +47,14 @@ def transform(self, blob: str) -> OscalBaseModel:
         """Transform the blob into a general OscalBaseModel."""
 
 
+class ProfileToNativeTransformer(TransformerBase):
+    """Abstract interface for transformers that specifically transform profile into Dict."""
+
+    @abstractmethod
+    def transform(self, profile: Profile) -> Dict:
```

Comment:
this type annotation violates the typing on the extension above.

My proposal would be we have a transformer base which is effectively def transform(self, object: Any) -> Any

then a forward transformer (blob to OSCAL) and a backwards transformer (OSCAL to blob) set of abstract classes.

With this in mind: I would make it a oscal -> string and string -> oscal set of interfaces which is a lower level than what you have done.

Reply:
If you want help with the abstractions let me know.